File: python/concurrency.py
import time
import multiprocessing

def worker():
    sum([i for i in range(1000000)])

if __name__ == "__main__":
    start = time.time()
    processes = []
    
    for _ in range(100):
        p = multiprocessing.Process(target=worker)
        p.start()
        processes.append(p)

    for p in processes:
        p.join()

    end = time.time()
    print(f"Time: {end - start} seconds")

File: typescript/concurrency.ts
import { Worker } from 'worker_threads';

const start = Date.now();
let completed = 0;

const workerScript = `
    const { parentPort } = require('worker_threads');
    let sum = 0;
    for (let i = 0; i < 1000000; i++) sum += i;
    parentPort.postMessage(sum);
`;

for (let i = 0; i < 100; i++) {
    const worker = new Worker(workerScript, { eval: true });
    worker.on('message', () => {
        completed++;
        if (completed === 100) {
            console.log(`Time: ${Date.now() - start} ms`);
        }
    });
    worker.on('error', (err) => {
        console.error(`Worker error: ${err}`);
    });
    worker.on('exit', (code) => {
        if (code !== 0) {
            console.error(`Worker stopped with exit code ${code}`);
        }
    });
}